### PR TITLE
Adding custom methods for LINQ queries

### LINKED ISSUE #3591: Adding custom methods for LINQ queries (C#)
### Proposed title of article
Adding custom methods for LINQ queries (C#)

### Introduction paragraph (2-3 paragraphs):
By introducing extension methods to the IEnumerableT> interface, you may expand the set of methods available for LINQ queries. You may, for example, develop a custom aggregate method to compute a single value from a sequence of data in addition to the usual average and maximum procedures. You can also build a method that returns a new series of values and acts as a custom filter or a specialized data transform for a sequence of variables. Distinct, Skip, and Reverse are examples of such techniques.

You may apply your own methods to any enumerable collection when you extend the IEnumerableT> interface.  In this article, we shall deeply take a look at various custom methods that are used in LINQ queries in C#.

### Key takeaways: 
1. Adding a method that returns a sequence
2. Overloading an Aggregate Method to Accept Various Types
3. Adding an aggregate method

### References: 
N/A

### PR DESCRIPTION
**Pull request by Joshuathuo**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #issue-number`
This closes #3591

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Good afternoon and thank you for submitting your article for publishing. @Joshuathuo 
Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. :rocket:
Thank you.

File: content/articles/adding-custom-methods-for-linq-queries/index.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
+### Adding custom methods for LINQ queries (C#) 
+### Introduction
+By introducing extension methods to the IEnumerableT> interface, you may expand the set of methods available for LINQ queries. Any data sequence can be converted into one single value by creating your custom aggregate method in SQL Server.
```
Comment (paulodhiambo) [inline content/articles/adding-custom-methods-for-linq-queries/index.md:3]: change "Any data sequence can be converted into one single value by creating your custom aggregate method in SQL Server. " to "You can convert any data sequence into one value by creating your custom aggregate method in SQL Server. "

File: content/articles/adding-custom-methods-for-linq-queries/index.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
+### Adding custom methods for LINQ queries (C#) 
+### Introduction
+By introducing extension methods to the IEnumerableT> interface, you may expand the set of methods available for LINQ queries. Any data sequence can be converted into one single value by creating your custom aggregate method in SQL Server. 
+
+You can also build a method that returns a new series of values and acts as a custom filter or a specialized data transform for a sequence of variables. 
+
+A few examples include the Distinct and Skip techniques, as well as the Reverse.
+
+You may apply your methods to any enumerable collection when you extend the IEnumerableT> interface. In this article, we shall deeply take a look at various custom methods that are used in LINQ queries in C#.
```
Comment (paulodhiambo) [inline content/articles/adding-custom-methods-for-linq-queries/index.md:9]: change "In this article, we shall deeply take a look at various custom methods that are used in LINQ queries in C#." to "In this article, we shall look at various custom methods used in LINQ queries in C#."

File: content/articles/adding-custom-methods-for-linq-queries/index.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+Console.WriteLine("double: Median = " + query1);
+/*
+ The output of the above code is as shown below
+
+ Double: Median = 5.95
+*/
+```
+### An aggregate method is overloaded by accepting a variety of types.
+To accept sequences of different types, you can overload your aggregate method. Overloading each type is the typical technique. It's also possible to use a delegate to build an overload for a generic type. Combining both approaches is also an option.
```
Comment (paulodhiambo) [inline content/articles/adding-custom-methods-for-linq-queries/index.md:59]: consider rewriting "It's also possible to use a delegate to build an overload for a generic type. Combining both approaches is also an option." to "However, it's also possible to use a delegate to build an overload for a generic type. Combining both approaches is also an option."

**Review by paulodhiambo — CHANGES_REQUESTED** [review]
@Joshuathuo, make the requested changes and ping me when done.

**Comment by Joshuathuo** [comment]
@paulodhiambo I have done the edits you can please have a look

**Comment by ahmadmardeni1** [comment]
plagiarism check done ✅
I think this article is light on substance - What do you think? @paulodhiambo

**Comment by Joshuathuo** [comment]
@paulodhiambo  may I know the status update on this PR kindly??

**Comment by paulodhiambo** [comment]
@Joshuathuo please consider explaining how to add the LINQ queries in a real application scenario.

**Comment by Joshuathuo** [comment]
A LINQ query is added say after maybe one has created a data source  and the a query is added as follows as an example

// Adding the query
//  "var cityQuery ="
IEnumerable<Cities> cityQuery =
    from city in cities
    where city.Population[0] > 10,000
    select city;

**Comment by Joshuathuo** [comment]
@paulodhiambo  please consider my comment above

**Comment by paulodhiambo** [comment]
@Joshuathuo your article is good but short, that's why I requested you to explain how we can use LINQ queries in a real application scenario. For example, try retrieving some data from the from an API like [JSON API](https://jsonplaceholder.typicode.com/) then apply the LINQ query to response, this would be more helpful.

**Comment by Joshuathuo** [comment]
@paulodhiambo  I have a question. Should I add it at the end of the article or where specifically should I add it

**Comment by Joshuathuo** [comment]
@paulodhiambo  I have added it you can have a look

**Comment by Joshuathuo** [comment]
I have just updated @paulodhiambo  you can have a look at it now

**Pull request merged**